services/src/main/java/org/exoplatform/addons/gamification/storage/dao/RuleDAO.java

```diff
@@ -95,6 +95,18 @@ public RuleEntity findRuleByEventAndDomain(String event, String domain) throws P
 
     }
 
+    public List<RuleEntity> getActiveRules() throws PersistenceException {
+
+        TypedQuery<RuleEntity> query = getEntityManager().createNamedQuery("Rule.getActiveRules", RuleEntity.class);
+
+        try {
+            return query.getResultList();
+        } catch (NoResultException e) {
+            return null;
```

Comment:
I think it is better to return an empty list in case of error, instead of null

Reply:
@rdenarie  IMO Returning null instead of empty list might result in better performance, because empty lists need time to be created and discarded, and they need space to be stored. To minimize this inconvenience, we should return shared immutable empty lists.

Reply:
You can return Collections.emptyList()

In fact, with an empty list, you dont create a special case to manage at upper level. It is clearer to read and clearer to use.

It is a development good practice, advised by Sonar.
You can find more informations here :
https://stackoverflow.com/questions/36779498/why-is-it-considered-good-practice-to-return-an-empty-collection